Shebang directive on top of the file breaks #r SDK parsing

In the Specifying an sdk u show an example creating a web app:

#r "sdk:Microsoft.NET.Sdk.Web"

using Microsoft.AspNetCore.Builder;

var a = WebApplication.Create();
a.MapGet("/", () => "Hello world");
a.Run();

====

I get error  saying 'AspNetCore' does not exist in namespace 'Microsoft'.

Has this changed?

environment:
Mac osx ARM
i have Core6 and Core7 Frameworks installed. Currently using Core7.

I created project using command as shown in doc $> dotnet script init

thx ~Marvin Foster
mmarvb7@gmail.com or marvinfoster@finsync.com

`➜ dotnet script index.csx --verbosity d --no-cache`
```
dbug: Dotnet.Script.Core.Commands.ExecuteScriptCommand[0]
      The script /Users/slavin/source/test/index.csx was executed with the '--no-cache' flag. Skipping cache.
info: Dotnet.Script.Core.ScriptCompiler[0]
      Current runtime is 'osx'.
dbug: Dotnet.Script.DependencyModel.ProjectSystem.ScriptProjectProvider[0]
      Creating project file for /Users/slavin/source/test/index.csx
dbug: Dotnet.Script.DependencyModel.ProjectSystem.ScriptParser[0]
      Parsing /Users/slavin/source/test/index.csx
dbug: Dotnet.Script.DependencyModel.ProjectSystem.ScriptProjectProvider[0]
      Project file saved to /Users/slavin/Library/Caches/dotnet-script/Users/slavin/source/test/net7.0/script.csproj
dbug: Dotnet.Script.DependencyModel.ProjectSystem.ScriptProjectProvider[0]
      <?xml version="1.0" encoding="utf-8"?>
      <Project Sdk="Microsoft.NET.Sdk">
        <PropertyGroup>
          <OutputType>Exe</OutputType>
          <TargetFramework>net7.0</TargetFramework>
          <LangVersion>latest</LangVersion>
        </PropertyGroup>
        <ItemGroup></ItemGroup>
        <Target Name="RecordReferencePaths" AfterTargets="AfterResolveReferences">
          <WriteLinesToFile File="$(OutputPath)/ReferencePaths.txt" Lines="@(ReferencePath)" />
        </Target>
      </Project>
dbug: Dotnet.Script.DependencyModel.Context.DotnetRestorer[0]
      Restoring /Users/slavin/Library/Caches/dotnet-script/Users/slavin/source/test/net7.0/script.csproj using the dotnet cli. RuntimeIdentifier : osx-x64 NugetConfigFile: /Users/slavin/.nuget/NuGet/NuGet.Config
dbug: Dotnet.Script.DependencyModel.Context.ProfiledRestorer[0]
      Restoring /Users/slavin/Library/Caches/dotnet-script/Users/slavin/source/test/net7.0/script.csproj took 1097ms
dbug: Dotnet.Script.Core.ScriptCompiler[0]
      Configuration/Optimization mode: Debug
/Users/slavin/source/test/index.csx(5,17): error CS0234: The type or namespace name 'AspNetCore' does not exist in the namespace 'Microsoft' (are you missing an assembly reference?)
/Users/slavin/source/test/index.csx(7,9): error CS0103: The name 'WebApplication' does not exist in the current context
```

--- a/src/Dotnet.Script.DependencyModel/ProjectSystem/ScriptParser.cs
+++ b/src/Dotnet.Script.DependencyModel/ProjectSystem/ScriptParser.cs
@@ -53,7 +53,7 @@ namespace Dotnet.Script.DependencyModel.ProjectSystem
         private static string ReadSdkFromReferenceDirective(string fileContent)
         {
             const string pattern = DirectivePatternPrefix + "r" + SdkDirectivePatternSuffix;
-            var match = Regex.Match(fileContent, pattern);
+            var match = Regex.Match(fileContent, pattern, RegexOptions.Multiline);
             if (match.Success)
             {
                 var sdk = match.Groups[1].Value;

--- a/src/Dotnet.Script.Tests/ScriptProjectProviderTests.cs
+++ b/src/Dotnet.Script.Tests/ScriptProjectProviderTests.cs
@@ -40,5 +40,17 @@ namespace Dotnet.Script.Tests
             var projectFileInfo = provider.CreateProject(TestPathUtils.GetPathToTestFixtureFolder("WebApi"), _scriptEnvironment.TargetFramework, true);
             Assert.Equal("Microsoft.NET.Sdk.Web", XDocument.Load(projectFileInfo.Path).Descendants("Project").Single().Attributes("Sdk").Single().Value);
         }
+
+        // See: https://github.com/dotnet-script/dotnet-script/issues/723
+        [Theory]
+        [InlineData("#!/usr/bin/env dotnet-script\n#r \"sdk:Microsoft.NET.Sdk.Web\"")]
+        [InlineData("#!/usr/bin/env dotnet-script\n\n#r \"sdk:Microsoft.NET.Sdk.Web\"")]
+        public void ShouldHandleShebangBeforeSdk(string code)
+        {
+            var parser = new ScriptParser(TestOutputHelper.CreateTestLogFactory());
+            var result = parser.ParseFromCode(code);
+
+            Assert.Equal("Microsoft.NET.Sdk.Web", result.Sdk);
+        }
     }
 }

